src/main/java/de/zalando/aruha/nakadi/controller/EventPublishingController.java

```diff
@@ -47,18 +42,27 @@ public EventPublishingController(final EventPublisher publisher, final EventType
     public ResponseEntity postEvent(@PathVariable final String eventTypeName, @RequestBody final String eventsAsString,
                                     final NativeWebRequest nativeWebRequest) {
         LOG.trace("Received event {} for event type {}", eventsAsString, eventTypeName);
+        final EventTypeMetrics eventTypeMetrics = eventTypeMetricRegistry.metricsFor(eventTypeName);
 
         try {
-            final EventTypeMetrics eventTypeMetrics = eventTypeMetricRegistry.metricsFor(eventTypeName);
+            final ResponseEntity response = postEventInternal(eventTypeName, eventsAsString, nativeWebRequest, eventTypeMetrics);
+            eventTypeMetrics.incrementResponseCount(response.getStatusCode().value());
+            return response;
+        } catch (RuntimeException ex) {
+            eventTypeMetrics.incrementResponseCount(-1); // yep, dunno response code here, related to event type.
```

Comment:
On this stage I can not say anything about response code, because it will be mapped later.
On the other side - InternalNakadiException creates 500 status code, and I need to separate it from unknown status code.